src/rpc/rawtransaction.cpp

```diff
@@ -917,14 +919,19 @@ UniValue sendrawtransaction(const JSONRPCRequest& request)
         );
 
     ObserveSafeMode();
+
+    CTransactionRef tx;
+    std::promise<void> promise;
+
+    { // cs_main scope
```

Comment:
Can you move this cs_main scope further down (to avoid having to declare the `CTransactionRef` above and then recompute the `GetHash()` at the end of this function?

Reply:
CTransactionRef->GetHash() is (virtually) free because we cache the hash in the CTransaction. Honestly we should probably just go the other way and remove the hashTx local. Additionally, if I'm correctly understanding your request, that would result in a deadlock as the validation interface queue may not be able to make progress as long as cs_main is held in the RPC thread. I've added a comment to validationinterface.h above CallFunctionInValidationInterfaceQueue to this effect.

Reply:
Misunderstood your point, moved the scope *start* down.

Reply:
Yes, sorry - my comment was unclear.

Change looks good.